src/istio/mixerclient/check_context.h

```diff
@@ -0,0 +1,197 @@
+      : shared_attributes_(shared_attributes), fail_open_(fail_open) {}
+
+  const istio::mixer::v1::Attributes* attributes() const {
+    return shared_attributes_->attributes();
+  }
+
+  const std::vector<istio::quota_config::Requirement>& quotaRequirements()
+      const {
+    return quota_requirements_;
+  }
+  std::vector<istio::quota_config::Requirement>& quotaRequirements() {
+    return quota_requirements_;
+  }
+
+  //
+  // Policy Cache Checks
+  //
+
+  bool policyCacheHit() const { return policy_cache_hit_; }
+  const google::protobuf::util::Status& policyStatus() const {
+    return policy_cache_result_.status();
+  }
+
+  void checkPolicyCache(CheckCache& policyCache) {
+    policyCache.Check(*shared_attributes_->attributes(), &policy_cache_result_);
+    policy_cache_hit_ = policy_cache_result_.IsCacheHit();
+  }
+
+  void updatePolicyCache(const google::protobuf::util::Status& status,
+                         const istio::mixer::v1::CheckResponse& response) {
+    policy_cache_result_.SetResponse(status, *shared_attributes_->attributes(),
+                                     response);
+  }
+
+  //
+  // Quota Cache Checks
+  //
+
+  bool quotaCheckRequired() const { return !quota_requirements_.empty(); }
+  bool remoteQuotaRequestRequired() const {
+    return remote_quota_check_required_;
+  }
+
+  void checkQuotaCache(QuotaCache& quotaCache) {
+    if (!quotaCheckRequired()) {
+      return;
+    }
+
+    //
+    // Quota is removed from the quota cache iff there is a policy cache hit. If
+    // there is a policy cache miss, then a request has to be sent upstream
+    // anyways, so the quota will be decremented on the upstream response.
+    //
+    quotaCache.Check(*shared_attributes_->attributes(), quota_requirements_,
+                     policyCacheHit(), &quota_cache_result_);
+
+    remote_quota_check_required_ =
+        quota_cache_result_.BuildRequest(allocRequestOnce());
+
+    quota_cache_hit_ = quota_cache_result_.IsCacheHit();
+  }
+
+  void updateQuotaCache(const google::protobuf::util::Status& status,
+                        const istio::mixer::v1::CheckResponse& response) {
+    quota_cache_result_.SetResponse(status, *shared_attributes_->attributes(),
+                                    response);
+  }
+
+  bool quotaCacheHit() const { return quota_cache_hit_; }
+  const google::protobuf::util::Status& quotaStatus() const {
+    return quota_cache_result_.status();
+  }
+
+  //
+  // Upstream request and response
+  //
+
+  void compressRequest(const AttributeCompressor& compressor,
+                       const std::string& deduplication_id) {
+    compressor.Compress(*shared_attributes_->attributes(),
+                        allocRequestOnce()->mutable_attributes());
+    request_->set_global_word_count(compressor.global_word_count());
+    request_->set_deduplication_id(deduplication_id);
+  }
+
+  bool failOpen() const { return fail_open_; }
```

Comment:
This seem to be inheriting value from `ClientContextBase::NetworkFailOpen()`, but uses a different name. Could you change one or the other to keep things consistent?

Reply:
will change to networkFailOpen()